### PR TITLE
[EMB-444][EMB-446][EMB-464] Sidenav, loading animation, registration form and then some

### PR DESCRIPTION
**Pull request by chrisseto**
## Purpose

* Implement the Registries SideNav component
* Implement the Registration Form View component
* Finish up most of the scaffolding of the overview page
* Initial work for getting anchors to work

## Side Effects

* Anchors/Fragment handling could mess with some things
* Probably some merge conflicts

## Feature Flags

N/A

## Ticket

https://openscience.atlassian.net/browse/EMB-444
https://openscience.atlassian.net/browse/EMB-446
https://openscience.atlassian.net/browse/EMB-464

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: lib/registries/addon/components/link/component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+@tagName('')
+@layout(template)
+export default class Link extends Component {
+    @service router!: RouterService;
+
+    route!: string;
+    models?: any[];
+    queryParams?: Record<string, any>;
+
+    tag: string = defaultTo(this.tag, 'a');
```
Comment (aaxelb) [inline lib/registries/addon/components/link/component.ts]: looks like `tag` isn't used

File: lib/registries/addon/components/link/component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import RouterService from '@ember/routing/router-service';
+import defaultTo from 'ember-osf-web/utils/default-to';
+import template from './template';
+
+type AnchorRel = 'noreferrer' | 'noopener';
+type AnchorTarget = '_self' | '_blank' | '_parent' | '_top';
+
+@tagName('')
+@layout(template)
+export default class Link extends Component {
```
Comment (aaxelb) [inline lib/registries/addon/components/link/component.ts]: This is my favorite custom link component so far (I think we're up to 4 or 5?). Maybe we can promote it and delete the others...

File: lib/registries/addon/components/side-nav/x-link/template.hbs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{{!-- NOTE: The duplication here is due to a bug/limitation of glimmer --}}
+{{!-- splattributes (...attributes) does not appear to work when rendering curried components --}}
+{{#if (or hasBlock @onclick)}}
+    <XDummy
+        @click={{action this.toggle}}
+        @tagName="a"
+        class={{local-class this.wrapperClasses}}
+        data-test-collapsed="{{this.collapsed}}"
```
[the remark this answers is not in the document]
Comment (chrisseto) [reply lib/registries/addon/components/side-nav/x-link/template.hbs]: Quotes force the boolean to cast to a string.
If you don't include the quotes the selector disappears if `this.collapsed` is false.

```hbs
<div data-test-collapsed="{{this.collapsed}}" />

{{!-- Can become --}}
<div data-test-collapsed="true" />
{{!-- Or --}}
<div data-test-collapsed="false" />



<div data-test-collapsed={{this.collapsed}} />

{{!-- Can become --}}
<div data-test-collapsed />
{{!-- Or --}}
<div />
````

File: mirage/config.ts [hunk 4]
```diff
@@ -33,12 +33,18 @@ export default function(this: Server) {
 
     osfResource(this, 'node', { except: ['create'] });
     this.post('/nodes/', createNode);
-    osfNestedResource(this, 'node', 'contributors');
-    osfNestedResource(this, 'node', 'linkedNodes', { only: ['index'] });
-    osfNestedResource(this, 'node', 'registrations', { only: ['index'] });
-    osfNestedResource(this, 'node', 'draftRegistrations', { only: ['index'] });
+    osfNestedResource(this, 'nodes', 'children');
```
[the remark this answers is not in the document]
Comment (chrisseto) [reply mirage/config.ts]: Fixin' 👍

File: lib/registries/addon/components/side-nav/x-link/template.hbs (hunk 3 above)
[the remark this answers is not in the document]
Comment (aaxelb) [reply lib/registries/addon/components/side-nav/x-link/template.hbs]: Ohh, good point. That's.... subtle.

**Comment by aaxelb** [comment]
Closing in favor of #496 (au revoir, Chris!)

**Comment by chrisseto** [comment]
🥈

**Pull request closed without merging**